Util_Environment.php

```diff
@@ -1064,8 +1067,8 @@ static public function detect_post_id() {
 			return $posts[0]->ID;
 		} elseif ( isset( $posts->ID ) ) {
 			return $posts->ID;
-		} elseif ( isset( $_REQUEST['p'] ) ) {
-			return (integer) $_REQUEST['p'];
+		} elseif ( isset( Util_Request::get_integer( 'p' ) ) {
```

Comment:
Use `! empty()`.  You may want to set a variable to avoid calling twice.

Reply:
Changed to ! empty and created a variable to avoid 2 function calls.